### PR TITLE
Update `buildspec` configuration

### PR DESCRIPTION
**Pull request by yusufsheiqh**
## Context

- Refactor: Remove `post_build` phase in `buildspec` configuration.
- Upgrade Python runtime version in `buildspec` configuration.
- Rename `requirements-test` -> `dev`.
- Add `requirements` folder.

## Additional information

JIRA ticket - https://uktrade.atlassian.net/browse/GPM-538

### CODE REVIEW

File: buildspec-tests.yml [hunk 1]
```diff
@@ -3,16 +3,12 @@ version: 0.2
 phases:
   install:
     runtime-versions:
-      python: 3.9
+      python: 3.11
     commands:
       - echo Installing dependencies...
-      - pip install -r requirements-test.txt
+      - pip install -r requirements/dev.txt
```
Comment (gabelton) [inline buildspec-tests.yml:9]: What's the thinking behind the rename?

**Review by gabelton — APPROVED** [review]

File: buildspec-tests.yml (hunk 1 above)
Comment (yusufsheiqh) [reply buildspec-tests.yml:9]: These requirements are mainly for developers and isn't only for testing purposes. The `pre-commit` hook is a dev dependency, for example.

Comment (lgarvey) [reply buildspec-tests.yml:9]: I suggest we move to poetry anyway. What do you think?

Comment (yusufsheiqh) [reply buildspec-tests.yml:9]: Yes - we have a JIRA ticket to address `poetry`: https://uktrade.atlassian.net/browse/GPM-530

**Pull request merged**